docs/decisions/0002-inter-app-apis.rst

```diff
@@ -0,0 +1,44 @@
+Inter-app APIs
+--------------
+
+Status
+======
+
+Accepted
+
+Context
+=======
+
+The edx-platform Django project is a conglomeration of different LMS and Studio features written in the structure of Django apps. Over the years, the boundaries between features have become muddled for various reasons. We now find apps intruding into the Python innards of other apps, making the intrusive apps tightly dependent on the inner behaviors of other apps.
+
+Due to this lack of clearly delimited separation of concerns, the monolith has become hard to maneuver, comprehend, and modify.
+
+Decisions
+=========
+
+#. Each Django app should clearly define a set of Python APIs that it exposes to other Django apps, which run within the LMS/Studio process.
+
+#. Each app's Python APIs should be defined/exported in a Python module called "api.py", located in the top-most directory of the app.
```

Comment:
@nasthagiri  Great to see the ADR process starting to address these issues!

I'd also like to see all python API declarations versioned from the beginning, so that they can be changed incrementally without disrupting dependencies.

For example, there's [two TODOs in the Grades API](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/grades/api.py#L19) already -- what should happen to the API when these are addressed?